flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/TestingSchedulingTopology.java

```diff
@@ -33,19 +32,23 @@
 import java.util.Map;
 import java.util.Optional;
 
+import static org.apache.flink.util.Preconditions.checkState;
+
 /**
  * A simple scheduling topology for testing purposes.
  */
 public class TestingSchedulingTopology
 	implements SchedulingTopology<TestingSchedulingExecutionVertex, TestingSchedulingResultPartition> {
 
+	private final List<TestingSchedulingExecutionVertex> verticesInAddedOrder = new ArrayList<>();
```

Comment:
Would it work to use a `LinkedHashMap` for `schedulingExecutionVertices`?

Reply:
Good suggestion! updated.